### PR TITLE
 NPE fix for System VM's start Command

### PR DESCRIPTION
**Pull request by pavanaravapalli**
## Description

System VM's are failed to deploy on VM ware setup, due to NPE. 

Solution : Added Null check for the same and took care of respective code flow changes. 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Yet to test and update.

### CODE REVIEW

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 1]
```diff
@@ -2286,7 +2286,7 @@ protected StartAnswer execute(StartCommand cmd) {
                 }
             }
 
-            if (!bootMode.equalsIgnoreCase(ApiConstants.BootType.BIOS.toString())) {
+            if (StringUtils.isNotBlank(bootMode) && !bootMode.equalsIgnoreCase(ApiConstants.BootType.BIOS.toString())) {
```
Comment (sureshanaparti) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: @pavanaravapalli  bootMode compared with boot type. Please check.

Comment (pavanaravapalli) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: @sureshanaparti 

here the purpose is to have bootMode initialise with default value and utilise it with uefi enabled vmprofile or normal vmprofile. however the constant which is used confusing so i have done changes and fixed.

**Comment by davidjumani** [comment]
Tested on VMware, LGTM

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by DaanHoogland** [comment]
@pavanaravapalli please review #4088/#4089

**Pull request merged**